hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmNegatedPredicate.java

```diff
@@ -58,7 +58,7 @@ public SqmPredicate getWrappedPredicate() {
 
 	@Override
 	public List<Expression<Boolean>> getExpressions() {
-		return Collections.singletonList( wrappedPredicate );
+		return Arrays.asList( wrappedPredicate );
```

Comment:
The list which `Arrays.asList` returns can not be added to unfortunately, so we need to use this:

```suggestion
		final List<Expression<Boolean>> list = new ArrayList<>( 1 );
		list.add( wrappedPredicate );
		return list;
```